include/chemfiles/Trajectory.hpp

```diff
@@ -195,7 +241,17 @@ class CHFL_EXPORT Trajectory final {
         return path_;
     }
 
+    /// Get a pointer to the memory addressed used for writing
+    ///
+    /// If the trajectory was not created for writing to memory, this will
+    /// return `nullopt`.
+    ///
+    /// @example{trajectory/memory_block.cpp}
+    optional<span<char>> memory_block() const;
```

Comment:
Should this return a std::string instead? 

And make it an empty string if the trajectory is not writing to memory?

Reply:
I think that returning a string isn't the best idea as `std::string` will make a deep copy of the buffer in this case. Maybe a blank `span<char>` or `span<const char>`?

Reply:
Oh, right. string_view would have been ideal then, but we don't use it in any other public header right now. So `span<const char>` is the best solution. I am not sure if returning an empty span instead of nullopt is the best thing to do: on one hand the type is nicer to work with, on the other hand we loose the distinction between not a memory trajectory and not yet written.

Reply:
I think `optional` makes sense here for semantic meaning. It allows one to determine if the `Trajectory` was opened to write to memory.